store/store.go

```diff
@@ -99,7 +99,7 @@ func (s *Store) Start() error {
 
 // Stop shuts down all of the working parts of the store.
 func (s *Store) Stop() error {
-	defer s.RunChannel.Close()
+	s.RunChannel.Close()
```

Comment:
Hopefully it is this simple.

In order to prevent existing job runs from accessing the db, we'll need to wait until the `JobRunner` has closed:
https://github.com/smartcontractkit/chainlink/blob/features/gui-bridges-page/services/job_runner.go#L90

It seems that it's closed due to `application.go` `Stop()`: https://github.com/smartcontractkit/chainlink/blob/features/gui-bridges-page/services/application.go#L86

But this is always closing asynchronously with no guarantee that it's stopped:
https://github.com/smartcontractkit/chainlink/blob/features/gui-bridges-page/services/job_runner.go#L114

Two ways to look at this:

1. This is good enough. Closing the channels doesn't mean the go routines have stopped but timing is on our side.
2. We should introduce `WaitGroup` waiters to ensure `JobRunner`'s worker channels actually close, blocking in the mean time, before leaving `Stop()`

Reply:
We could also save this for a subsequent commit.

Reply:
I was thinking about the WaitGroup, but I think it could be difficult to pull off and so this would at least make some of the failures on CI less likely in the meantime.